From 461b57c8147972d01f06a819844f00768433591f Mon Sep 17 00:00:00 2001
From: Zhang Meng <zhangmeng.kevin@spacemit.com>
Date: Wed, 4 Feb 2026 08:54:40 +0800
Subject: [PATCH 357/450] RVCK: driver: clk: k3: keep some system based clock
 always on

FROM: https://github.com/RVCK-Project/rvck/pull/213

community inclusion
category: bugfix
bugzilla: https://github.com/RVCK-Project/rvck/issues/212

--------------------------------

The hdma clk is used by some component of CCI bus, it should
be keep always on, regardless of whether hdma enabled.

The rcpu clk should be always on because it is running backround.

Signed-off-by: Zhang Meng <zhangmeng.kevin@spacemit.com>
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 drivers/clk/spacemit/ccu-k3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/spacemit/ccu-k3.c b/drivers/clk/spacemit/ccu-k3.c
index 764426359540c..6acde3b76b7b9 100644
--- a/drivers/clk/spacemit/ccu-k3.c
+++ b/drivers/clk/spacemit/ccu-k3.c
@@ -866,7 +866,7 @@ static const struct clk_parent_data rcpu_clk_parents[] = {
 	CCU_PARENT_HW(pll1_d6_409p6),
 };
 CCU_MUX_DIV_GATE_FC_DEFINE(rcpu_clk, rcpu_clk_parents, APMU_RCPU_CLK_RES_CTRL,
-			   4, 3, BIT(15), 7, 3, BIT(12), 0);
+			   4, 3, BIT(15), 7, 3, BIT(12), CLK_IS_CRITICAL);
 
 static const struct clk_parent_data dsi4ln2_dsi_esc_parents[] = {
 	CCU_PARENT_HW(pll1_d48_51p2_ap),
-- 
2.55.0

